src/plugins/workspace/public/components/workspace_picker_content/workspace_picker_content.tsx

```diff
@@ -27,31 +33,51 @@ const getValidWorkspaceColor = (color?: string) =>
   validateWorkspaceColor(color) ? color : undefined;
 
 interface Props {
+  searchQuery: string;
   coreStart: CoreStart;
   registeredUseCases$: BehaviorSubject<WorkspaceUseCase[]>;
   onClickWorkspace?: () => void;
 }
 
 export const WorkspacePickerContent = ({
+  searchQuery,
   coreStart,
   registeredUseCases$,
   onClickWorkspace,
 }: Props) => {
   const workspaceList = useObservable(coreStart.workspaces.workspaceList$, []);
+  const isDashboardAdmin = coreStart.application.capabilities?.dashboards?.isDashboardAdmin;
   const availableUseCases = useObservable(registeredUseCases$, []);
 
-  const filteredWorkspaceList = useMemo(() => {
-    return workspaceList.slice(0, MAX_WORKSPACE_PICKER_NUM);
-  }, [workspaceList]);
-
   const filteredRecentWorkspaces = useMemo(() => {
     return recentWorkspaceManager
       .getRecentWorkspaces()
       .map((workspace) => workspaceList.find((ws) => ws.id === workspace.id))
-      .filter((workspace): workspace is WorkspaceObject => workspace !== undefined)
-      .slice(0, MAX_WORKSPACE_PICKER_NUM);
+      .filter((workspace): workspace is WorkspaceObject => workspace !== undefined);
   }, [workspaceList]);
 
+  const queryedWorkspace = useMemo(() => {
+    if (searchQuery) {
+      const normalizedQuery = searchQuery.toLowerCase();
+      const result = workspaceList.filter((item) => {
+        return item.name.toLowerCase().indexOf(normalizedQuery) > -1;
```

Comment:
Maybe we could extract query logic into a new function like queryFromList(list,query), then we don't need to use same query logic twice.